cmd/login/flag.go

```diff
@@ -68,7 +70,13 @@ func (f *flag) Init(cmd *cobra.Command) {
 	cmd.Flags().BoolVar(&f.ClusterAdmin, flagClusterAdmin, false, "Log in as Giant Swarm staff member.")
 	cmd.Flags().BoolVar(&f.InternalAPI, flagInternalAPI, false, "Use Internal API in the kube config. Please check the documentation for more details.")
 	cmd.Flags().StringVar(&f.SelfContained, flagSelfContained, "", "Create a self-contained kubectl config with embedded credentials and write it to this path.")
-	cmd.Flags().BoolVar(&f.KeepContext, flagKeepContext, false, "Keep the current kubectl context. If not set/false, will set the current-context to the one representing the cluster to log in to.")
+
+	f.KeepContextEnvVar = os.Getenv("KUBECTL_GS_LOGIN_KEEP_CONTEXT")
```

Comment:
You can actually use `viper` to simplify this.
For example in mcli: https://github.com/giantswarm/mcli/blob/e98ea44d44178d097e74bee93593ad93af50d2f7/cmd/root_flags.go#L51-L53

Reply:
Refactored :)